### PR TITLE
organic repair surgery can be done on a cyborg torso

### PR DESCRIPTION
**Pull request by TheEaterOfBeans**
## About The Pull Request

Title says it all.

## Why It's Good For The Game

Sometimes people need their torsos replaced due to infection, resulting in someone who has a cyborg torso but is otherwise organic.

## Changelog
:cl:
tweak: you can repair organic limbs on a person with a cyborg torso
/:cl:

### CODE REVIEW

File: code/modules/surgery/healing.dm [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 	target_mobtypes = list(/mob/living/carbon/human, /mob/living/carbon/monkey)
 	possible_locs = list(BODY_ZONE_CHEST)
-	requires_bodypart_type = BODYPART_ORGANIC
+	requires_bodypart_type = 0
```
Comment (silicons) [inline code/modules/surgery/healing.dm]: NONE instead of 0, please!

**Review by DeltaFire15 — CHANGES_REQUESTED** [review]
To prevent some issues (mainly, confusion for people who see both surgery types pop up on people with robot limbs, even if a full robot), please add an check akin to repair surgery for robots, just with the robot limb check replaced with a check for if there's at least one organic limb.
 (reference: ../code/modules/surgery/robot_healing.dm, proc: /datum/surgery/robot_healing/can_start)

**Comment by TheEaterOfBeans** [comment]
> To prevent some issues (mainly, confusion for people who see both surgery types pop up on people with robot limbs, even if a full robot), please add an check akin to repair surgery for robots, just with the robot limb check replaced with a check for if it's an organic limb.
> (reference: ../code/modules/surgery/robot_healing.dm, proc: /datum/surgery/robot_healing/can_start)

...how would I check if OTHER limbs are organic? This is for people with mechanical torsos and an otherwise organic body.

**Comment by DeltaFire15** [comment]
> > To prevent some issues (mainly, confusion for people who see both surgery types pop up on people with robot limbs, even if a full robot), please add an check akin to repair surgery for robots, just with the robot limb check replaced with a check for if it's an organic limb.
> > (reference: ../code/modules/surgery/robot_healing.dm, proc: /datum/surgery/robot_healing/can_start)
> 
> ...how would I check if OTHER limbs are organic? This is for people with mechanical torsos and an otherwise organic body.

Look at the reference I mentioned, it checks for if at least one limb is robotic and only does the other cheks / allows it if that'd be a case. here it'd basically just be the inverted version, aka checking for if at least one bodypart is overganic (ignoring hybrid bodyparts, so with the TRUE arg for the is_organic proc

The code part here is:
(of course with different intendation, look at the reference for that)
>/datum/surgery/robot_healing/can_start(mob/user, mob/living/carbon/target, obj/item/tool)
>    for(var/obj/item/bodypart/B in target.bodyparts)
>		if(B.is_robotic_limb())
>			return ..()

Copypaste it into a reasonable place in the file, with the correct path (so the normal heal surgery's instead of robot_healing), change the is_robotic_limb() to is_organic_limb(TRUE), and it should work

**Review by DeltaFire15 — APPROVED** [review]

**Pull request merged**